web/src/beta/features/Editor/Settings/hooks.ts

```diff
@@ -0,0 +1,67 @@
+import { useState, useEffect } from "react";
+
+import { NLSLayer } from "@reearth/services/api/layersApi/utils";
+import { Page } from "@reearth/services/api/storytellingApi/utils";
+
+type SettingProps = {
+  onPageUpdate?: (id: string, layers: string[]) => void;
+  layers?: NLSLayer[];
+  selectedPage?: Page;
+};
+export default ({ layers, selectedPage, onPageUpdate }: SettingProps) => {
+  const [layerChecked, setLayerChecked] = useState<string[]>([]);
+  const [allLayersChecked, setAllLayersChecked] = useState(false);
+
+  useEffect(() => {
+    if (selectedPage && layers) {
+      const selectedLayerIds = selectedPage.layersIds || [];
+      setLayerChecked(selectedLayerIds);
+    }
+  }, [selectedPage, layers]);
+
+  useEffect(() => {
+    if (selectedPage && layers) {
+      const allLayerIds = layers.map(layer => layer.id);
+      const allLayersSelected =
+        layerChecked.length === allLayerIds.length &&
+        layerChecked.every(layerId => allLayerIds.includes(layerId));
+      setAllLayersChecked(allLayersSelected);
+    }
+  }, [selectedPage, layers, layerChecked]);
```

Comment:
I think you can do this simpler if you just make `allLayersChecked` a useMemo. Something like this (probably incorrect or needs some tweaking, but hopefully you get the idea):
```
const allLayersChecked = useMemo(()=> layers && checkedLayers && (layers == checkedLayers),[layers, checkedLayers]);
```